File: address-book/external-contacts/address-book-synchronization/code-samples/syncAddressBook.py
# https://developers.ringcentral.com/my-account.html#/applications
# Find your credentials at the above url, set them as environment variables, or enter them below

# PATH PARAMETERS
accountId = '<ENTER VALUE>'
extensionId = '<ENTER VALUE>'

# OPTIONAL QUERY PARAMETERS
queryParams = {
    #'syncType': [ 'FSync', 'ISync' ],
    #'syncToken': '<ENTER VALUE>',
    #'perPage': 000,
    #'pageId': 000
}

import os
from ringcentral import SDK
rcsdk = SDK(os.environ['clientId'], os.environ['clientSecret'], os.environ['serverURL'])
platform = rcsdk.platform()
platform.login(os.environ['username'], os.environ['extension'], os.environ['password'])
r = platform.get(f'/restapi/v1.0/account/{accountId}/extension/{extensionId}/address-book-sync', queryParams)
# PROCESS RESPONSE

File: address-book/external-contacts/address-book-synchronization/code-samples/syncAddressBook.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
string accountId = "<ENTER VALUE>";
string extensionId = "<ENTER VALUE>";

// OPTIONAL QUERY PARAMETERS
SyncAddressBookParameters syncAddressBookParameters = new SyncAddressBookParameters {
    //syncType = new[] { "FSync", "ISync" },
    //syncToken = "<ENTER VALUE>",
    //perPage = 000,
    //pageId = 000
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Account(accountId).Extension(extensionId).AddressBookSync().Get(syncAddressBookParameters);
// PROCESS RESPONSE
